### PR TITLE
[interpreter] Fix native as entry point

### PR DESCRIPTION
**Pull request by tnowacki**
## Motivation

- interpreter entrypoint did not handle the case where the function is native

## Test Plan

- by value test, by ref test, type test

### CODE REVIEW

File: language/move-vm/runtime/src/interpreter.rs [hunk 1]
```diff
@@ -101,15 +102,45 @@ impl Interpreter {
...
+                    Some(id) => e
+                        .at_code_offset(function.index(), 0)
+                        .finish(Location::Module(id.clone())),
+                    None => PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
+                        .with_message(
+                            "Unexpected native function not located in a module".to_owned(),
+                        )
+                        .finish(Location::Undefined),
+                })?;
+            Ok(return_values.into_iter().collect())
```
Comment (amnn) [inline language/move-vm/runtime/src/interpreter.rs:138]: Is there some equivalent to the paranoid checks we do in the other code path that should go here?

**Review by amnn — APPROVED** [review]
Seems reasonable!

File: language/move-vm/runtime/src/interpreter.rs [hunk 2]
```diff
@@ -101,15 +102,45 @@ impl Interpreter {
...
+        if function.is_native() {
+            for arg in args {
+                interpreter
+                    .operand_stack
+                    .push(arg)
+                    .map_err(|e| e.finish(Location::Undefined))?;
+            }
+            let resolver = function.get_resolver(loader);
+            let return_values = interpreter
+                .call_native_return_values(
```
Comment (dariorussi) [inline language/move-vm/runtime/src/interpreter.rs:120]: just out of curiosity... I have a PR where I moved collecting the return values from the stack in this function.
If we had that we could have called directly the `call_native` as it was and fetch the value from the stack... or not?

**Review by oxade — APPROVED** [review]
good find

**Review by dariorussi — APPROVED** [review]
it does feel like we should review native invocation some day....

File: language/move-vm/runtime/src/interpreter.rs (hunk 2 above)
Comment (tnowacki) [reply language/move-vm/runtime/src/interpreter.rs:120]: Ah, yeah we could retrieve them from the stack. Do you want me to attempt to do that in this PR?

**Pull request merged**